### PR TITLE
Run integration tests in CI

### PR DESCRIPTION
**Pull request by pankajkoti**
closes: #198 
closes: #206

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -104,10 +104,110 @@ jobs:
...
+
+      - name: Install packages and dependencies
+        run: |
+          python -m pip install uv
+          uv pip install --system hatch
+          hatch -e tests.py${{ matrix.python-version }}-${{ matrix.airflow-version }} run pip freeze
+
+      - name: Test DAG Factory against Airflow ${{ matrix.airflow-version }} and Python ${{ matrix.python-version }}
+        run: |
+          PYTHONPATH=`pwd`:`pwd`/examples CONFIG_ROOT_DIR=`pwd`"/dags" hatch run tests.py${{ matrix.python-version }}-${{ matrix.airflow-version }}:test-integration-setup
```
Comment (tatiana) [inline .github/workflows/cicd.yaml:185]: Would it make sense to set
```
PYTHONPATH: $PWD:$PWD/examples
```
Or
```
PYTHONPATH: .:./examples
```

In the `env` section (currently line 187)? Same for `CONFIG_ROOT_DIR`

**Review by tatiana — APPROVED** [review]
This looks great, @pankajkoti , really happy we'll have these integration tests to give us more confidence when making changes and accepting contributions to DAG factory.

I left a minor feedback inline.

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -104,10 +104,110 @@ jobs:
...
+            airflow-version: "2.4"
+          - python-version: "3.12"
+            airflow-version: "2.5"
+          - python-version: "3.12"
+            airflow-version: "2.6"
+          - python-version: "3.12"
+            airflow-version: "2.7"
+          - python-version: "3.12"
+            airflow-version: "2.8"
+    services:
```
Comment (pankajastro) [inline .github/workflows/cicd.yaml:148]: Does our test use this Postgres sevice?

**Review by pankajastro — APPROVED** [review]

File: .github/workflows/cicd.yaml (hunk 1 above)
Comment (pankajkoti) [reply .github/workflows/cicd.yaml]: yes, thanks. setting & using env vars now. $PWD din't seem to evaluate well. I used ${{ github.workflow }} and that seems to work well.

File: .github/workflows/cicd.yaml (hunk 2 above)
Comment (pankajkoti) [reply .github/workflows/cicd.yaml]: good check, yes, we don't need postgres service, cleaned it up :)

**Pull request merged**